com.openfoam.OpenFOAM.yaml

```diff
@@ -0,0 +1,74 @@
+app-id: com.openfoam.OpenFOAM
+runtime: org.freedesktop.Platform
+sdk: org.freedesktop.Sdk
+runtime-version: 24.08
+command: openfoam
+finish-args:
+  - --share=ipc
+  - --device=dri
+  - --filesystem=home
+modules:
+  - name: openmpi
+    buildsystem: autotools
+    config-opts:
+      - --enable-mpi-cxx
+      - --disable-wrapper-runpath
+    sources:
+      - type: archive
+        url: https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.4.tar.bz2
+        sha256: 92912e175fd1234368c8730c03f4996fe5942e7479bb1d10059405e7f2b3930d
+
+  - name: OpenFOAM
+    build-options:
+      env:
+        GMP_ARCH_PATH: /usr
+        MPFR_ARCH_PATH: /usr
+        LDFLAGS: "-lmpfr -lgmp"
+    buildsystem: simple
+    build-commands:
+      - cat applications/utilities/mesh/generation/foamyMesh/conformalVoronoiMesh/Make/options
+      - bash -lc "source etc/bashrc && cd ThirdParty && ls -l && ./Allwmake -j$(nproc) && ./makeAdios2"
+      - bash -lc "source etc/bashrc && ./Allwmake -j$(nproc)"
+      - install -d /app/OpenFOAM
+      - cp -a . /app/OpenFOAM
+      - install -d /app/bin
+      - |
+        cat <<'EOF' > /app/bin/openfoam
+        #!/usr/bin/env bash
+        # OpenFOAM runtime environment
+        source /app/OpenFOAM/etc/bashrc
+        if [ $# -eq 0 ]; then
+          exec /usr/bin/bash -i
+        else
+          exec "$@"
+        fi
+        EOF
```

Comment:
Never use shell commands to generate a script. Use `type: script` or directly a file.